### ISSUE TITLE
Keycard not recognized on Linux (Manjaro)

### ISSUE THREAD

**Issue by Artiom-M** [status-im/status-desktop#16768]
# Bug Report

## Description

Desktop app doesn't see the card reader

## Steps to reproduce

Start App fresh
Select "I already use Status"
Select "Login with keycard"
Screen display:
![Image](https://github.com/user-attachments/assets/ae6d414c-fd76-4a49-8ad0-0fe0ada94e21)

### Additional Information
Card and reader works with keycard_cli and pcscd_scan
- Status desktop version:
StatusIm-Desktop-v2.31.1-a322b9-x86_64.AppImage
Log looks like:
```
INFO [11-19|03:47:18.127] Status backend initialized               backend=geth version=3.4.0-5-ga01c7ed1e commit=a01c7ed1e IpfsGatewayURL=https://ipfs.status.im/
INF 2024-11-19 03:47:18.376Z app info                                   topics="status-app" tid=751391 file=nim_status_client.nim:246 version=2.31.1 commit=a322b99d7 currentDateTime=2024-11-19T14:47:18+11:00
INF 2024-11-19 03:47:18.376Z starting application controller...         topics="status-app" tid=751391 file=nim_status_client.nim:248
INF 2024-11-19 03:47:20.409Z starting application...                    topics="status-app" tid=751391 file=nim_status_client.nim:251
WRN 2024-11-19 03:47:20.409Z Error decoding signal                      topics="signals-manager" tid=751391 file=signals_manager.nim:49 err="Unknown signal received: mediaserver.started"
keycard - no-pcsc
INF 2024-11-19 03:47:34.859Z shutting down...                           topics="status-app" tid=751391 file=nim_status_client.nim:222
INF 2024-11-19 03:47:34.859Z logging out...                             topics="app-controller" tid=751391 file=app_controller.nim:296
ERROR[11-19|03:47:34.864|github.com/status-im/status-go/mobile/status.go:958] error in makeJSONResponse                error="there is no running node"
```
- Operating System:
Manjaro Linux

pcsc log:
```
Nov 19 14:58:17 pure systemd[1]: Started PC/SC Smart Card Daemon.
Nov 19 14:58:17 pure (pcscd)[752857]: pcscd.service: Referenced but unset environment variable evaluates to an empty string: PCSCD_ARGS
Nov 19 14:58:18 pure pcscd[752857]: 00000000 ../PCSC/src/winscard_svc.c:404:ContextThread() Communication protocol mismatch!
Nov 19 14:58:18 pure pcscd[752857]: 00000016 ../PCSC/src/winscard_svc.c:406:ContextThread() Client protocol is 4:4
Nov 19 14:58:18 pure pcscd[752857]: 00000002 ../PCSC/src/winscard_svc.c:408:ContextThread() Server protocol is 4:5
```

**Comment by anastasiyaig** [comment]
@Artiom-M hey thanks for reporting. In order to make keycard operational on Linux there is a need to install additional lib 

```
sudo apt-get update
sudo apt-get -y install pcscd
```

**Comment by Artiom-M** [comment]
I have pcscd working. There are logs from it in my report.

**Comment by anastasiyaig** [comment]
alright, can you pls share which linux you are using just now?

**Comment by Artiom-M** [comment]
It's Manjaro, as written on the report.

**Comment by anastasiyaig** [comment]
@caybro @saledjenic maybe you guys can take a look. It works to me on Ubuntu 22.04 just fine

**Comment by Drewsapple** [comment]
@Artiom-M I know it's not recommended on arch, but if you aren't using pcscd for other things, downgrading the pcsclite package to 2.2.3 allowed status to communicate with my reader.

**Comment by bitgamma** [comment]
@saledjenic cannot un-assign myself, but this is not solvable in status-keycard-go from what I read. I think that the version of pcsclite-lib linked when building the appimage for linux needs to be updated (and tested to remain compatible with systems with an older pcscd)

**Comment by anastasiyaig** [comment]
@siddarthkay @jakubgs maybe you can help?

**Comment by Artiom-M** [comment]
Downgrading PCSC will be complicated. However if this is a version issue, could the error message be more user friendly?

**Comment by Parveshdhull** [comment]
I am also facing the same issue in nixos

```
00760747 [139754075150016] ../src/winscard_svc.c:404:ContextThread() Communication protocol mismatch!
00000021 [139754075150016] ../src/winscard_svc.c:406:ContextThread() Client protocol is 4:4
00000003 [139754075150016] ../src/winscard_svc.c:408:ContextThread() Server protocol is 4:5
```

**Comment by guylouis** [comment]
lots of customers are coming back to us with this issue, can someone ELI5 to me what we can do on our status desktop side?

**Comment by siddarthkay** [comment]
I'll take a quick stab at bundling `pcsclite` version `2.2.3` with the current appImage and see if this fixes the issue.

**Comment by siddarthkay** [comment]
an initial attempt was made here : https://github.com/status-im/status-desktop/pull/17382
The built appImage for linux is https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-250226-035422-29ea99-pr17382-x86_64.tar.gz
Please try it out and report back if this fixes the keycard not recognized issue.

**Comment by Parveshdhull** [comment]
hi @siddarthkay, Thank you for working on this.

I tried this build but getting same error "PCSC Not available"

Status desktop logs:
```
loading "libpcsclite_real.so.1" failed: libpcsclite_real.so.1: cannot open shared object file: No such file or directory
keycard - no-pcsc
keycard - no-pcsc
```

pscsd logs: No longer showing [this error](https://github.com/status-im/status-desktop/issues/16768#issuecomment-2585649156), just running log.
`systemd[1]: Started PC/SC Smart Card Daemon`

**Comment by siddarthkay** [comment]
Thanks for taking a look @Parveshdhull 
I tried to use the keycard flow on a fresh Ubuntu and I see this 

<img width="438" alt="Image" src="https://github.com/user-attachments/assets/27bde12e-bf78-4362-ad95-831d9d73cee9" />

So clearly the bundling needs work on my end.
I'll spend some more time on this today.

**Comment by siddarthkay** [comment]
With https://github.com/status-im/status-desktop/pull/17382/commits/bbc209aec563643e0099507eee792d9f2db18d53 i got this error message to go away `loading "libpcsclite_real.so.1" failed: libpcsclite_real.so.1: cannot open shared object file: No such file or directory`
But I still need to do more work to fix  `keycard - no-pcsc`

**Comment by siddarthkay** [comment]
bundling `pcscd` along with `pcslite` in the AppImage seems to be working.

Now when I run status-desktop it refers to the library from the App Image and not my system environment.
I still need to clean up this PR : https://github.com/status-im/status-desktop/pull/17382 and fix nix builds but I consider this a good success.

![Image](https://github.com/user-attachments/assets/80b2324e-7ca2-437e-b32a-c70dee9b6bcc)

**Comment by siddarthkay** [comment]
@Artiom-M : can you please check if this build works for you on Linux Manjaro? https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-250323-083638-50b745-pr17382-x86_64.tar.gz

**Comment by Artiom-M** [comment]
I can see the plug in reader screen, I plug in and it asks to insert the card. And when I put the card on it :

![Image](https://github.com/user-attachments/assets/40486a2b-daac-4344-ad9c-146cab6ba64c)
Log:
```

29270926 [123343766685376] ccid_usb.c:2162:Multi_PollingProc() 1/103: Slot 0: 0x00
00031135 [123343667717824] ifdhandler.c:1437:IFDHPowerICC() action: PowerUp, usb:072f/2200:libudev:0:/dev/bus/usb/001/103 (lun: 0)
00000500 [123343667717824] ../src/eventhandler.c:404:EHStatusHandlerThread() powerState: POWER_STATE_POWERED
00000020 [123343667717824] ../src/eventhandler.c:421:EHStatusHandlerThread() Card inserted into ACS ACR122U 00 00
00000011 [123343667717824] ../src/winscard_svc.c:855:MSGSignalClient() Signal client: 15
00000007 [123343667717824] ../src/winscard_svc.c:858:MSGSignalClient() SIGNAL for client 15, rv=SCARD_S_SUCCESS
00000039 [123343667717824] Card ATR: 3B 85 80 01 80 73 C8 21 10 0E 
00000158 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000016 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000081 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000015 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000096 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_GET_READERS_STATE from client 15
00000150 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_GET_READERS_STATE from client 15
00000126 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000012 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000080 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000013 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000096 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CONNECT from client 15
00000015 [123343676110528] ../src/winscard_svc.c:526:ContextThread() Authorized client for 'ACS ACR122U 00 00'
00000008 [123343676110528] ../src/winscard.c:253:SCardConnect() Attempting Connect to ACS ACR122U 00 00 using protocol: 3
00000007 [123343676110528] ../src/readerfactory.c:832:RFReaderInfo() RefReader() count was: 1
00000007 [123343676110528] ../src/winscard.c:347:SCardConnect() powerState: POWER_STATE_IN_USE
00000007 [123343676110528] ../src/prothandler.c:108:PHSetProtocol() Attempting PTS to T=1
00000009 [123343676110528] ifdhandler.c:889:IFDHSetProtocolParameters() protocol T=1, usb:072f/2200:libudev:0:/dev/bus/usb/001/103 (lun: 0)
00000009 [123343676110528] ../src/winscard.c:428:SCardConnect() Active Protocol: T=1
00000015 [123343676110528] ../src/winscard.c:454:SCardConnect() hCard Identity: 2c80b3d9
00000011 [123343676110528] ../src/winscard.c:516:SCardConnect() UnrefReader() count was: 2
00000013 [123343676110528] ../src/winscard_svc.c:546:ContextThread() CONNECT for client 15, rv=SCARD_S_SUCCESS
00000350 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: TRANSMIT from client 15
00000022 [123343676110528] ../src/readerfactory.c:859:RFReaderInfoById() RefReader() count was: 1
00000008 [123343676110528] ../src/winscard.c:1597:SCardTransmit() Send Protocol: T=1
00000017 [123343676110528] ifdhandler.c:1669:IFDHTransmitToICC() usb:072f/2200:libudev:0:/dev/bus/usb/001/103 (lun: 0)
00102649 [123343676110528] ../src/winscard.c:1650:SCardTransmit() UnrefReader() count was: 2
00000037 [123343676110528] ../src/winscard_svc.c:719:ContextThread() TRANSMIT for client 15, rv=SCARD_S_SUCCESS
00001430 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: TRANSMIT from client 15
00000054 [123343676110528] ../src/readerfactory.c:859:RFReaderInfoById() RefReader() count was: 1
00000014 [123343676110528] ../src/winscard.c:1597:SCardTransmit() Send Protocol: T=1
00000012 [123343676110528] ifdhandler.c:1669:IFDHTransmitToICC() usb:072f/2200:libudev:0:/dev/bus/usb/001/103 (lun: 0)
00073071 [123343676110528] ../src/winscard.c:1650:SCardTransmit() UnrefReader() count was: 2
00000044 [123343676110528] ../src/winscard_svc.c:719:ContextThread() TRANSMIT for client 15, rv=SCARD_S_SUCCESS
keycard - Pair failed invalid card cryptogram
00050979 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000011 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001103 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000007 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000100 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000007 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000102 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000006 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00500958 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000040 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001249 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000033 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000229 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000224 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00500886 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000039 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001200 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000191 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000185 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501030 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000045 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001126 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000022 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000089 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000016 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000081 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000017 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501011 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000068 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001149 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000180 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000033 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000184 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501182 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000044 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001226 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000032 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000177 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000026 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000175 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000025 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00500989 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000053 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001180 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000062 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000078 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000022 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000157 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000021 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501084 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000044 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001207 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000030 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000184 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000029 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000141 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000019 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501011 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000043 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00001261 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000032 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00000225 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15
00000033 [123343676110528] ../src/winscard_svc.c:867:MSGSendReaderStates() Send reader states: 15
00000212 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_STOP_WAITING_READER_STATE_CHANGE from client 15
00000032 [123343676110528] ../src/winscard_svc.c:465:ContextThread() CMD_STOP_WAITING_READER_STATE_CHANGE for client 15, rv=SCARD_S_SUCCESS
00501311 [123343676110528] ../src/winscard_svc.c:381:ContextThread() Received command: CMD_WAIT_READER_STATE_CHANGE from client 15

```
last messages repeat forever.

Couple obseravtions: 
1. I have to unplug my Nitrokey in order it to recognize the actual reader
2. I have to unplug the reader and plug back when on "Plug in your card reader page" in order it to be recognized
3. New profile with empty keykard doesn't work at all, it doesn't see the card reader.

**Comment by siddarthkay** [comment]
perhaps @bitgamma can make sense of these logs?

**Comment by bitgamma** [comment]
> keycard - Pair failed invalid card cryptogram
this is normal if the pairing password is not the default one, in that case the app should ask for the pairing password

the other logs look like the software is scanning for reader/cards. Let's summon @igor-sirotin, maybe he has more details

**Comment by Artiom-M** [comment]
It didn't ask and the password is definitely not the default one.

**Comment by igor-sirotin** [comment]
> this is normal if the pairing password is not the default one, in that case the app should ask for the pairing password

Unfortunately, custom pairing passwords are not yet supported by status-desktop

**Comment by jakubgs** [comment]
My understanding is that the infra side of this work is done, and the ball is in the Desktop court. I'm marking it done in our project.
